LNX-59 Add player vision

alpinus4 opened the pull request:
Add 'look' action that returns list of objects in a set direction. 
Then player can iterate over such list and make decisions on what to do.

Sample code:
```python
objects = look(Direction.NORTH)
if len(objects) == 1 and "Floor" in objects:
    move(Direction.NORTH)
```
Only move when there's nothing beside floor.

It is to be discussed whether checking objects by string is acceptable or it should be changed to some type of enum.
Checking by string seemed to be the easiest and least problematic.

I added `self.name = self.__class__.__name__` to every object.
It holds it's name as string.
Can be further customized per object if class name is inappropriate to be seen by player.


Sample output:
```
{ "base_class_name" : "Action", "class_name" : "NextScene", "properties" : {} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 0, "position": "(0,0)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 1, "position": "(0,1)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 2, "position": "(1,0)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 3, "position": "(1,1)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 4, "position": "(0,2)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 5, "position": "(1,2)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 6, "position": "(0,3)"} } 
{ "base_class_name" : "Object", "class_name" : "Floor", "properties" : {"id": 7, "position": "(1,3)"} } 
{ "base_class_name" : "Object", "class_name" : "TrainingDummy", "properties" : {"id": 8, "position": "(0,3)"} } 
{ "base_class_name" : "Object", "class_name" : "TrainingDummy", "properties" : {"id": 9, "position": "(1,0)"} } 
{ "base_class_name" : "Object", "class_name" : "Agent", "properties" : {"id": 10, "position": "(0,0)"} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 8} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 9} } 
{ "base_class_name" : "Action", "class_name" : "Look", "properties" : {"object_id": 10, "direction": "NORTH"} } 
{ "base_class_name" : "Action", "class_name" : "Move", "properties" : {"object_id": 10, "direction": "NORTH"} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 8} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 9} } 
{ "base_class_name" : "Action", "class_name" : "Look", "properties" : {"object_id": 10, "direction": "NORTH"} } 
{ "base_class_name" : "Action", "class_name" : "Move", "properties" : {"object_id": 10, "direction": "NORTH"} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 8} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 9} } 
{ "base_class_name" : "Action", "class_name" : "Look", "properties" : {"object_id": 10, "direction": "NORTH"} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 8} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 9} } 
{ "base_class_name" : "Action", "class_name" : "Look", "properties" : {"object_id": 10, "direction": "NORTH"} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 8} } 
{ "base_class_name" : "Action", "class_name" : "Idle", "properties" : {"object_id": 9} } 
```

alpinus4 committed:
Add player vision

Add 'look' action that returns list of objects in a set direction.
Then player can iterate over such list and make decisions on what to do.

alpinus4 committed:
Remove redundant attribute declarations

blazej-smorawski commented:
Have you though about returning reference or sth? Though it may not be possible to use it in players code without some additional imports. It's just a though, returning string is fine but you could also add `id`, so the player could later use the received `id` to perform some actions.

alpinus4 committed:
Change the return type of look to dictionary

Keys are object names, and id's  are values.
With that we can still check if string is in objects.

alpinus4 commented:
I changed the return type of look action to dictionary with such structure:
```python
"name" : 1 #id
```

now player can get id of an object by simply

```python
objects = look(Direction.NORTH)
if "Enemy" in objects:
    enemy_id = objects["Enemy"]
    # do sth based on id
```

but can still perform checks for specific type of object, so it's not overly complicated
```python
if "Enemy" in objects:
```
